### PR TITLE
add prog basicMorphoFilter

### PR DESCRIPTION
**Pull request by kerautret**

### CODE REVIEW

**Comment by dcoeurjo** [comment]
I think you must detail which structural element your are using for the morpho. Furthermore, if you consider disk structural elements, please use Euclidean distance transformation for Erode/Dilate, it will be way more efficient (and multithread...).

**Comment by kerautret** [comment]
yes I add it, it is just a basic square element, agree as Tristan do in the tuto for sphere perhaps later ;)

File: geometry3d/basicMorphoFilter.cpp [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+  try{
+    po::store(po::parse_command_line(argc, argv, general_opt), vm);
+  }catch(const std::exception& ex){
+    trace.info()<< "Error checking program options: "<< ex.what()<< std::endl;
+    parseOK=false;
+  }
+  po::notify(vm);
+  if(vm.count("help")||argc<=1|| !parseOK )
+  {
+    trace.info()<< "Apply basic morpho filter from a cubical structurant element" <<std::endl << "Options: "<<std::endl
```
Comment (dcoeurjo) [inline geometry3d/basicMorphoFilter.cpp]: structurAL element

**Comment by dcoeurjo** [comment]
Even with a basic cubical structural element, you can use the DistanceTransformation with l_infty norm and it will be more efficient.

**Comment by dcoeurjo** [comment]
(and I really dislike not rotational invariant S.E....) :)

**Comment by kerautret** [comment]
agree for binary image but here it also works on gray level images (and contains also a median filter)
So perhaps we could do a bw version based on DT later ;)

File: geometry3d/basicMorphoFilter.cpp [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+  unsigned int cpt = 0;
+  
+  for(ImageDomIterator it = anImage.domain().begin(); it != anImage.domain().end(); ++it){
+    trace.progressBar(cpt, imageSize);
+    Point3D pt = *it;
+    std::vector<unsigned char> vectVal;
+    for (int k = -(int)size; k <= (int) size; k++) {
+      for (int l = -(int)size; l <= (int) size; l++) {
+        for (int m = -(int)size; m <= (int) size; m++) {
+          Point3D p (((int)(pt[0]))+k, (int)(pt[1])+l, ((int)(pt[2]))+m);
```
Comment (dcoeurjo) [inline geometry3d/basicMorphoFilter.cpp]: ok for the DT but could you please just add a test here to make sure that p is in the ball ?
e.g. ``` (p-pt).norm()<size``` 

I think it is important to not have squares structural elements..

**Comment by kerautret** [comment]
yes thanks done

**Comment by kerautret** [comment]
ping@dcoeurjo  can I merge ?

**Comment by dcoeurjo** [comment]
Fine. Thanks.

**Pull request merged**